File: 04-polimorfismo/python/triangulo.py
"""
triangulo.py

Hereda de Figura pero NO de Dibujable: no firmo ese contrato.
Compara la primera linea de la clase con la de circulo.py.
"""

from figura import Figura


class Triangulo(Figura):

    def __init__(self, base, altura):
        super().__init__("Triangulo")
        self._base = base
        self._altura = altura

    def area(self):
        return self._base * self._altura / 2

    def perimetro(self):
        return self._base * 3            # simplificacion: equilatero

File: 04-polimorfismo/java/Triangulo.java
/*
 * Triangulo.java
 *
 * Esta clase hereda de Figura, pero NO implementa Dibujable:
 * no firmo ese contrato. Compara el encabezado con el de Circulo.java.
 */
public class Triangulo extends Figura {

    private double base, altura;

    public Triangulo(double base, double altura) {
        super("Triangulo");
        this.base = base;
        this.altura = altura;
    }

    @Override
    public double area() {
        return base * altura / 2;
    }

    @Override
    public double perimetro() {
        return base * 3;   // simplificacion: lo tratamos como equilatero
    }
}
